extensions/VisualEditor/wikia/ApiAddMediaTemporary.php

```diff
@@ -154,6 +156,14 @@ protected function checkPermissions() {
 		}
 	}
 
+	protected function checkPermissionsForJA() {
```

Comment:
You could just have ```if(local is japan && !wgdisableanonymouseditibg) { ... } else { checkPermissions(); }```

Btw. In general I like idea of small single responsibility functions/methods, however having value of having value of config setting (ja) in function name is a big red flag to me.